### DISCUSSION TITLE
How do I rename or move a directory managed by pipenv?

### DISCUSSION THREAD

**Discussion by JDLH** [pypa/pipenv#5123] [Q&A] [answered]
I am developing a small Python utility. I have a project directory, call it `~/2022/Foo/`. I manage this directory with **pipenv**.

How can I change the name of this directory, say to be `~/2022/Bar/`, and not mess up **pipenv**'s management of it? How can I move this directory to another parent, say to `~/2023/Foo/`?

I see that **pipenv** creates magical stuff in `$HOME/.local/share/virtualenvs/`, e.g. `Foo-1E7o8dGZ`. If I rename or move my project directory, I don't see a reason why that would change this magical stuff. But would changing the names maybe break the connection from this stuff to the project directory?

**Comment by J-M0** [comment, marked as answer]
When pipenv creates an venv for your project, it creates a hash of the path of your project as part of the name of the venv; that's the `1E7o8dGZ` in your `Foo-1E7o8dGZ`. If you rename the folder, pipenv will lose the connection to the venv and end up making a new one. To make sure you don't have an extra venv lying around, you can run `pipenv --rm` before renaming your project and then simply `pipenv install` to recreate it.

**Reply by JDLH** [reply]
Thank you for the very helpful answer. I think this should be in the documentation. I created issue #5129 to request adding a section saying this to the documentation.

**Reply by coolove** [reply]
May I know what bundle the project directory and the virtual environment directory? Is the hash inside the _meta section in Pipfile.lock file? when pipenv shell command run, how pipenv locate the correct virtual environment to switch? Where those mapping information stored? Thanks in advance.

**Reply by rohlay** [reply]
> When pipenv creates an venv for your project, it creates a hash of the path of your project as part of the name of the venv; that's the `1E7o8dGZ` in your `Foo-1E7o8dGZ`. If you rename the folder, pipenv will lose the connection to the venv and end up making a new one. To make sure you don't have an extra venv lying around, you can run `pipenv --rm` before renaming your project and then simply `pipenv install` to recreate it.

To further elaborate on this answer, does this imply _you can't rename or move a directory_ managed by Pipenv? As you have said, the only way of doing this (renaming or moving) is by _removing the environment and creating a new one_. 

Is it possible, for example, to simply change the absolute path in the `.project` file to the new location of the `Pipfile`? I am not talking about using `PIPENV_PIPFILE`, but rather _relocating_ the `Pipfile`.
